bioblend/galaxy/datasets/__init__.py

```diff
@@ -151,18 +226,54 @@ def get_datasets(self, limit=500, offset=0, history_id=None):
           For example, if ``limit`` is set to 100 and ``offset`` to 200,
           datasets 200-299 will be returned.
 
-        :type history_id: str
-        :param history_id: Encoded history ID to filter on.
+        :type user_details: bool
+        :param user_details: If ``True`` and the user is an admin, add the user
+          email to each returned dataset dictionary.
 
-        :rtype: list
-        :return: Return a list of dataset dicts.
+        .. note::
+          The following filtering options can only be used with Galaxy ``release_21.05`` or later:
+            workflow_id, invocation_id, job_id, dataset_collection_id, user_id,
+            date_range_min, date_range_max, user_details
         """
-        params = {
+        params: Dict[str, Any] = {
             'limit': limit,
             'offset': offset,
         }
         if history_id:
             params['history_id'] = history_id
+
+        q: List[str] = []
+        qv: List[Any] = []
```

Comment:
Why not just append to `params` like with `history_id`?

Reply:
It does not work.

The options `limit`, `offset` and `history_id` are defined as parameters of the `index` endpoint method and can therefore be included directly in `params`. 

The other options are expected as `q={filter-option}-{operator}` and `qv={value}` pairs which are then parsed into filters on the server.

Reply:
Using a `filters` parameter containing all `q`, `qv` options would be cleaner, but also less self-documenting.

```
class Filter:
    def __init__(self, param: str, val: Any, op: str = 'eq'):
        self.q = f'{param}-{op}'
        self.qv = val if op != 'in' else ','.join(val)
```

```
def get_datasets(..., filters: Optional[List[Filter]] = None) -> list:
    ...
    
    if filters:
        params['q'] = [f.q for f in filters] 
        params['qv'] = [f.qv for f in filters]
    
    return self._get(params=params)
```

```
datasets = self.gi.get_datasets(..., filters=[
    Filter('extension', 'txt'),
    Filter('state', ['ok', 'queued'], op='in'),
    ...
])
```